core/src/replay_stage.rs

```diff
@@ -2210,72 +2235,127 @@ impl ReplayStage {
     ) -> bool {
         let mut did_complete_bank = false;
         let mut tx_count = 0;
-        let mut execute_timings = ExecuteTimings::default();
-        let active_banks = bank_forks.read().unwrap().active_banks();
-        trace!("active banks {:?}", active_banks);
 
-        for bank_slot in &active_banks {
-            // If the fork was marked as dead, don't replay it
-            if progress.get(bank_slot).map(|p| p.is_dead).unwrap_or(false) {
-                debug!("bank_slot {:?} is marked dead", *bank_slot);
+        // Make shared structures thread safe.
+        let progress = RwLock::new(progress);
+        let longest_replay_time_us = AtomicU64::new(0);
+        let blockstore = RwLock::new(blockstore);
+        let my_pubkey = RwLock::new(my_pubkey);
+        let vote_account = RwLock::new(vote_account);
+        let transaction_status_sender = RwLock::new(transaction_status_sender);
+        let verify_recyclers = RwLock::new(verify_recyclers);
+        let replay_vote_sender = RwLock::new(replay_vote_sender);
+        let transaction_cost_metrics_sender = RwLock::new(transaction_cost_metrics_sender);
```

Comment:
Why do we need to wrap all these in `RwLock`?
Like doesn't seem to `my_pubkey` is mutated. 
Or sender channels are already thread-safe. They just need to be `clone`d for each thread.

Reply:
We don't. `clone` works wayyyy better here. Fixed